docs/tutorials/essentials/part-7-rtk-query-basics.md

```diff
@@ -0,0 +1,692 @@
+  baseQuery: fetchBaseQuery({ baseUrl: '/fakeApi' }),
+  // The "endpoints" represent operations and requests for this server
+  endpoints: builder => ({
+    // The `getPosts` endpoint is a "query" operation that returns data
+    getPosts: builder.query({
+      // The URL for the request is '/fakeApi/posts'
+      query: () => '/posts'
+    })
+  })
+})
+
+// Export the auto-generated hook for the `getPost` query endpoint
+export const { useGetPostsQuery } = apiSlice
+```
+
+RTK Query's functionality is based on a single method, called `createApi`. All of the Redux Toolkit APIs we've seen so far are UI-agnostic, and could be used with _any_ UI layer. The RTK Query core logic is the same way. However, RTK Query also includes a React-specific version of `createApi`, and we need to use that. So, we import from `'@reduxjs/toolkit/query/react'` specifically.
+
+#### API Slice Parameters
+
+When we call `createApi`, there are three fields that are required:
+
+- `reducerPath`: defines the expected top-level state slice field. For our other slices like `postsSlice`, there's no guarantee that it will be used to update `state.posts` - we _could_ have attached the reducer anywhere in the root state, like `someOtherField: postsReducer`. Here, `createApi` expects us to tell it where the cache state will exist when we add the cache reducer to the store.
+- `baseQuery`: a function that knows how to fetch data from the server. RTK Query includes `fetchBaseQuery`, a small wrapper around the standard `fetch()` function that handles typical processing of requests and responses. When we create a `fetchBaseQuery` instance, we can pass in the base URL of all future requests, as well as override behavior such as modifying request headers.
+- `endpoints`: a set of operations that we've defined for interacting with this server. Endpoints can be **_queries_**, which return data for caching, or **_mutations_**, which send an update to the server. The endpoints are defined using a callback function that accepts a `builder` parameter and returns an object containing endpoint definitions created with `builder.query()` and `builder.mutation()`.
+
+#### Defining Endpoints
+
+The first part of the URL for all requests is defined as `'/fakeApi'` in the `fetchBaseQuery` definition.
+
+For our first step, we want to add an endpoint that will return the entire list of posts from the fake API server. We'll include an endpoint called `getPosts`, and define it as a **query endpoint** using `builder.query()`. This method accepts many options for configuring how to make the request and process the response. For now, all we need to do is supply the remaining piece of the URL path by defining a `query` option, with a callback that returns the URL string: `() => '/posts'`.
+
+By default, query endpoints will use a `GET` HTTP request, but you can override that by returning an object like `{url: '/posts', method: 'POST', body}` instead of just the URL string itself.
```

Comment:
since `body` is not an established variable, maybe `body: { name: "Tom" }` or something?